libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -240,11 +363,14 @@ Builder& Builder::triangleCount(size_t triangleCount) noexcept {
 }
 
 Builder& Builder::triangles(const uint3* triangle32) noexcept {
+    ASSERT_PRECONDITION(!mMesh->mInput->triangles16,
+            "Triangles already provided in unsigned shorts");
     mMesh->mInput->triangles32 = triangle32;
     return *this;
 }
 
 Builder& Builder::triangles(const ushort3* triangle16) noexcept {
+    ASSERT_PRECONDITION(!mMesh->mInput->triangles32, "Triangles already provided in unsigned ints");
```

Comment:
because this is noexcept you can't have a `ASSERT_PRECONDITON`.

The pattern that I used elsewhere is to not check for any errors in the builder's method, and do all the error checking in `build`, which is the only method to not be `noexcept`.

Reply:
done